dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs

```diff
@@ -571,79 +571,74 @@ private static (Type ReturnType, Func<Kernel, KernelFunction, object?, ValueTask
-                return new FunctionResult(function, taskResult, kernel.Culture);
+                    var taskResult = Invoke(asTaskResultGetter, task, Array.Empty<object>());
+                    return new FunctionResult(function, taskResult, kernel.Culture);
+                }
+                );
             }
-            );
-        }
 
-        // IAsyncEnumerable<T>
-        if (returnType.GetGenericTypeDefinition() is Type genericAsyncEnumerable && genericAsyncEnumerable == typeof(IAsyncEnumerable<>))
-        {
-            Type elementType = returnType.GetGenericArguments()[0];
+            // IAsyncEnumerable<T>
+            if (returnType.GetGenericTypeDefinition() is Type genericAsyncEnumerable && genericAsyncEnumerable == typeof(IAsyncEnumerable<>))
+            {
+                Type elementType = returnType.GetGenericArguments()[0];
 
-            MethodInfo? getAsyncEnumeratorMethod = typeof(IAsyncEnumerable<>)
-                .MakeGenericType(elementType)
-                .GetMethod("GetAsyncEnumerator");
+                MethodInfo? getAsyncEnumeratorMethod = typeof(IAsyncEnumerable<>)
+                    .MakeGenericType(elementType)
+                    .GetMethod("GetAsyncEnumerator");
 
-            if (getAsyncEnumeratorMethod is not null)
-            {
-                return (returnType, (kernel, function, result) =>
+                if (getAsyncEnumeratorMethod is not null)
                 {
-                    var asyncEnumerator = Invoke(getAsyncEnumeratorMethod, result, s_cancellationTokenNoneArray);
-
-                    if (asyncEnumerator is not null)
+                    return (returnType, (kernel, function, result) =>
                     {
-                        return new ValueTask<FunctionResult>(new FunctionResult(function, asyncEnumerator, kernel.Culture));
-                    }
+                        var asyncEnumerator = Invoke(getAsyncEnumeratorMethod, result, s_cancellationTokenNoneArray);
+
+                        if (asyncEnumerator is not null)
+                        {
+                            return new ValueTask<FunctionResult>(new FunctionResult(function, asyncEnumerator, kernel.Culture));
+                        }
 
-                    return new ValueTask<FunctionResult>(new FunctionResult(function));
+                        return new ValueTask<FunctionResult>(new FunctionResult(function));
+                    }
+                    );
                 }
-                );
             }
         }
 
-        // Unrecognized return type.
-        throw GetExceptionForInvalidSignature(method, $"Unknown return type {returnType}");
+        // For everything else, just use the result as-is.
+        return (returnType, (kernel, function, result) =>
+        {
+            return new ValueTask<FunctionResult>(new FunctionResult(function, result, kernel.Culture));
+        }
+        );
```

Comment:
Small formatting suggestion:
```suggestion
        });
```